src/api/src/config.rs

```diff
@@ -4,119 +4,109 @@
-            as_addr: None,
-            as_config_file_path: None,
-            #[cfg(feature = "amber-as")]
-            amber: None,
-            #[cfg(feature = "policy")]
-            policy_path: None,
-        }
-    }
+    /// HTTPS session timeout in minutes.
+    pub timeout: i64,
+
+    /// HTTPS private key.
+    pub private_key: Option<PathBuf>,
+
+    /// HTTPS Certificate.
+    pub certificate: Option<PathBuf>,
+
+    /// Insecure HTTP.
+    /// WARNING: Using this option makes the HTTP connection insecure.
+    pub insecure_http: bool,
+
+    /// Public key used to authenticate the resource registration endpoint token (JWT).
+    /// Only JWTs signed with the corresponding private keys are authenticated.
+    pub auth_public_key: Option<PathBuf>,
+
+    /// Insecure HTTP APIs.
+    /// WARNING: Using this option enables KBS insecure APIs such as Resource Registration without
+    /// verifying the JWK.
+    pub insecure_api: bool,
+
+    /// Policy engine configuration used for evaluating whether the TCB status has access to
+    /// specific resources.
+    #[cfg(feature = "policy")]
+    pub policy_engine_config: Option<PolicyEngineConfig>,
 }
 
-impl TryFrom<&Path> for Config {
-    /// Load `Config` from a JSON configuration file like:
-    ///    {
-    ///        "repository_type": "LocalFs",
-    ///        "repository_description": {
-    ///            "dir_path": "/opt/confidential-containers/kbs/repository"
-    ///        },
-    ///            "attestation_token_type": "CoCo",
-    ///        # Only used in Remote Attestation-Service mode
-    ///        "as_addr": "http://127.0.0.1:50004",
-    ///        # Only used in Native Attestation-Service mode
-    ///        "as_config_file_path": "/etc/as-config.json",
-    ///        # Only used in Amber Attestation-Service mode
-    ///        "amber" : {
-    ///            "base_url": "https://amber.com",
-    ///            "api_key": "tBfd5kKX2x9ahbodKV1...",
-    ///            "certs_file": "/etc/amber/amber-certs.txt",
-    ///            # Optional, default is false.
-    ///            "allow_unmatched_policy": true
-    ///        },
-    ///        "policy_path": "/opt/confidential-containers/kbs/policy.rego"
-    ///    }
+impl TryFrom<&Path> for KbsConfig {
     type Error = anyhow::Error;
+
+    /// Load `Config` from a configuration file. Supported formats are all formats supported by the
+    /// `config` crate. See `KbsConfig` for schema information.
     fn try_from(config_path: &Path) -> Result<Self, Self::Error> {
-        let file = File::open(config_path)
-            .map_err(|e| anyhow!("failed to open KBS config file {}", e.to_string()))?;
+        let c = Config::builder()
+            .set_default("attestation_token_type", "CoCo")?
+            .set_default("sockets", vec!["127.0.0.1:8080"])?
+            .set_default("insecure_api", false)?
+            .set_default("insecure_http", false)?
```

Comment:
Would it be slightly cleaner to define all of these as constants?

Reply:
I assume you mean define the *values* as constants and not the keys. I've pushed new changes as follows:

* I've used the default value of `AttestationTokenVerifierType` for `attestation_token_type`.
* I've added a constant for the default socket address.
* The two `insecure_` values are bools. Does it make sense to you to use constants for these, too?
* The `timeout` value was already a constant.

Reply:
Could put the bools as constant just for completeness but I suspect we're not going to be changing those values anytime soon.

Reply:
Done. I've used constants for the bools, too.

I resorted to adding a string constant for `attestation_token_type` because using the default from `AttestationTokenVerifierType` is tricky due to conditional compilation.